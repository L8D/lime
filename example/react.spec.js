const lime = require('../lib/lime')
const mapValues = require('lodash.mapvalues')
const {makeExampleStory} = require('./suite')
const conditions = require('./conditions')
const properties = require('./properties')

describe('example suite results', () => {
  const mockProperties = mapValues(properties, jest.fn)
  let results

  beforeAll(() => {
    return lime.run([
      makeExampleStory({
        properties: mockProperties,
        conditions
      })
    ]).then((r) => {
      results = r
    })
  })

  it('should be defined', () => {
    expect(results).toBeDefined()
  })

  it('should have results', () => {
    lime.logResults(results)
  })

  it('should fire all the events', () => {
    expect(mockProperties.haveMinimumHits.mock.calls).toMatchSnapshot();
  });
})
