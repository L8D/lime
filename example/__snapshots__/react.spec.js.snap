// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example suite results should fire all the events 1`] = `
Array [
  Array [
    5,
  ],
]
`;
